configure.ac

```diff
@@ -87,81 +64,125 @@ AC_ARG_WITH([state-file-path],
         ;;
       *)
-        STATE_FILE_PATH="$withval"
+	STATEFILE_val="$withval"
         ;;
     esac
   ])
-AC_DEFINE_UNQUOTED([STATEFILE], "$STATE_FILE_PATH")
-AC_SUBST(STATE_FILE_PATH)
+AS_IF([test x$STATEFILE_val != x], [
+  state_file_path="$withval"
+], [
+  AS_CASE([${host_os}],
+    [*hpux*],
+      [state_file_path="/var/run/logrotate.status"],
+    [state_file_path="/var/lib/logrotate.status"]
+ )
+])
+AC_DEFINE_UNQUOTED([STATEFILE], ["$state_file_path"], [State file path.])
+AC_SUBST([STATE_FILE_PATH], [$state_file_path])
 
+
+DEFAULT_MAIL_COMMAND=""
 AC_ARG_WITH([default-mail-command],
   AC_HELP_STRING([--with-default-mail-command=COMMAND],
-                 [default mail command (e.g. /bin/mail -s)]),
+                 [mail command, default path: see configure output, often /usr/bin/mail]),
   [
     case "$withval" in
       yes|no)
         AC_MSG_ERROR([--with-default-mail-command=COMMAND requires a command to be specified])
         ;;
       *)
         DEFAULT_MAIL_COMMAND="$withval"
+        AC_DEFINE_UNQUOTED([DEFAULT_MAIL_COMMAND], "$DEFAULT_MAIL_COMMAND")
+        AC_SUBST(DEFAULT_MAIL_COMMAND)
         ;;
     esac
   ])
-AC_DEFINE_UNQUOTED([DEFAULT_MAIL_COMMAND], "$DEFAULT_MAIL_COMMAND")
-AC_SUBST(DEFAULT_MAIL_COMMAND)
+AS_IF([test "x$DEFAULT_MAIL_COMMAND" = x], [
+  AS_CASE([${host_os}],
+    [*hpux*],
+      [AC_CHECK_PROG([MAIL_COMMAND], [mailx], [$as_dir/$ac_word$ac_exec_ext], [], [/usr/bin:/bin])],
+    [*solaris*],
+      [AC_CHECK_PROG([MAIL_COMMAND], [mailx], [$as_dir/$ac_word$ac_exec_ext], [], [/usr/bin:/bin])],
+    [*netbsd*],
+      [AC_CHECK_PROG([MAIL_COMMAND], [mail], [$as_dir/$ac_word$ac_exec_ext -s], [], [/usr/bin:/bin])],
+    [AC_CHECK_PROG([MAIL_COMMAND], [mail], [$as_dir/$ac_word$ac_exec_ext], [], [/usr/bin:/bin])]
+  )
+  AS_IF([test "x$ac_cv_prog_MAIL_COMMAND" != 'x'], [
+    default_mail_comm=$ac_cv_prog_MAIL_COMMAND
+  ], [
+    default_mail_comm="/bin/mail"
+  ])
+  AC_DEFINE_UNQUOTED([DEFAULT_MAIL_COMMAND], ["$default_mail_comm"], [Mail command path.])
+  AC_SUBST([DEFAULT_MAIL_COMMAND], [$default_mail_comm])
+])
 
+COMPRESS_COMMAND_val=''
 AC_ARG_WITH([compress-command],
-  AC_HELP_STRING([--with-compress-command=COMMAND],
-                 [compress command (default: /bin/gzip)]),
+  [AC_HELP_STRING([--with-compress-command=COMMAND], [compress command, default path: see configure output, often /usr/bin/gzip])],
   [
     case "$withval" in
       yes|no)
         AC_MSG_ERROR([--with-compress-command=COMMAND requires a command to be specified])
         ;;
       *)
-        COMPRESS_COMMAND="$withval"
+        COMPRESS_COMMAND_val="$withval"
         ;;
     esac
   ])
-AC_DEFINE_UNQUOTED([COMPRESS_COMMAND], "$COMPRESS_COMMAND")
-AC_SUBST(COMPRESS_COMMAND)
+AS_IF([test "x$COMPRESS_COMMAND_val" = x], [
+  AC_CHECK_PROG([COMPRESS_COMMAND], [gzip], [$as_dir/$ac_word$ac_exec_ext], [], [/usr/bin:/bin:/usr/contrib/bin])
```

Comment:
I do not think I like this commit because it assumes that all the programs are installed on the build system, which is often not the case.  We should strictly distinguish build-time and run-time dependencies.